pkg/api/validation/validation.go

```diff
@@ -1242,6 +1242,29 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldForbidden("data", "Maximum secret size exceeded"))
 	}
 
+	switch secret.Type {
```

Comment:
Can't this just be replaced with a TODO to add future validations based on types?